modules/distr-tracing-tempo-install-cli.adoc

```diff
@@ -0,0 +1,231 @@
+
+.Procedure
+
+. Install the {TempoOperator}:
+
+.. Create an OpenShift project for the {TempoOperator}.
++
+[source,terminal]
+----
+$ oc apply -f - << EOF
+apiVersion: project.openshift.io/v1
+kind: Project
+metadata:
+  labels:
+    kubernetes.io/metadata.name: openshift-tempo-operator
+    openshift.io/cluster-monitoring: "true"
+  name: openshift-tempo-operator
+EOF
+----
+
+.. Create an Operator group.
++
+[source,terminal]
+----
+$ oc apply -f - << EOF
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: openshift-tempo-operator
+  namespace: openshift-tempo-operator
+spec:
+  upgradeStrategy: Default
+EOF
+----
+
+.. Create a subscription.
++
+[source,terminal]
+----
+$ oc apply -f - << EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: tempo-product
+  namespace: openshift-tempo-operator
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: tempo-product
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+EOF
+----
+
+.. Check the Operator status.
++
+[source,terminal]
+----
+$ oc get csv -n openshift-tempo-operator
+----
+
+. Run `oc apply -f` for a YAML file or standard input to create a Secret for your object storage bucket:
++
+[source,terminal]
+----
+$ oc apply -f <secret_file>.yaml
+----
++
+[source,terminal]
+----
+$ oc apply -f - << EOF
+<object_storage_secret>
+EOF
+----
++
+--
+include::snippets/distr-tracing-tempo-required-secret-parameters.adoc[]
+--
++
+--
+include::snippets/distr-tracing-tempo-secret-example.adoc[]
+--
+
+. Run `oc new-project` to create an OpenShift project of your choice for the *TempoStack* instance that you will create in the next step.
++
+[TIP]
+====
+[source,terminal]
+----
+$ oc new-project <openshift_project_of_tempostack_instance>
+----
+
+[source,terminal]
+----
+$ oc apply -f - << EOF
+apiVersion: project.openshift.io/v1
+kind: Project
+metadata:
+  name: <openshift_project_of_tempostack_instance>
+EOF
+----
+====
+
+. Create a *TempoStack* instance in the OpenShift project that you created for the *TempoStack* instance in the previous step.
++
+NOTE: You can create multiple *TempoStack* instances in separate projects on the same cluster.
++
+.. Customize the `TempoStack` custom resource (CR):
++
+[source,yaml]
+----
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+metadata:
+  name: sample
+  namespace: <openshift_project_of_tempostack_instance>
+spec:
+  storageSize: 1Gi
+  storage:
+      secret:
+          name: <secret-name> <1>
+          type: <secret-provider> <2>
+  template:
+    queryFrontend:
+      jaegerQuery:
+        enabled: true
+        ingress:
+          route:
+            termination: edge
+          type: route
+----
+<1> The value of the `name` in the `metadata` of the Secret.
+<2> The accepted values are `azure` for Azure Blob Storage, `gcs` for Google Cloud Storage, and `s3` for Amazon S3 or MinIO or Red Hat OpenShift Data Foundation.
```

Comment:
Is `s3` for all three that follow it? Might consider an unordered list of these to help keep them straight. In any case, if it is for all three suggest the serial comma format:

```suggestion
<2> The accepted values are `azure` for Azure Blob Storage, `gcs` for Google Cloud Storage, and `s3` for Amazon S3, MinIO, or Red Hat OpenShift Data Foundation.
```

You could also use `;` since there is internal punctuation in one of the items:
```suggestion
<2> The accepted values are `azure` for Azure Blob Storage; `gcs` for Google Cloud Storage; and `s3` for Amazon S3, MinIO, or Red Hat OpenShift Data Foundation.
```

Reply:
Thank you so much. I've used your second suggestion with `;` and applied it manually.